### ISSUE TITLE
reproduce it on tensorflow

### ISSUE THREAD

**Issue by ln23415** [layumi/Person_reID_baseline_pytorch#125]
hi~, I try reproduce your model on tensorflow1.13.3, I use the same networks structure that Fine-tune Resnet_v1_50(exclude the 1000 Logits classifier layer), add the 512 nodes fully connected layer as feature layer then connect with the 751 nodes fully connect layer as  the classifier.Both the 512 feature layer and 751 classifier layer are use None activation function, use [mean : 0 , stddev : 0.0625]  and  [mean : 0 , stddev : 0.001] to initialize the weights at the feature layer and classifier layer.But the accuracy on classification (72%) is far away from your classification result(99.8%) in train.py.
the *.tfrecords files are processed in same way as prepare.py.
Do anyone know why?
this is my code
`# -*- coding: UTF-8 -*-
import os
import numpy as np
import tensorflow as tf
import tensorflow.contrib.slim as slim

import tensorflow.contrib.slim.python.slim.nets.resnet_v1 as resnet_v1
import json

os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
os.environ["CUDA_VISIBLE_DEVICES"] = "0, 1"
train_tfrecords = "./train.tfrecords"
val_tfrecords = "./val.tfrecords"

TRAIN_FILE = "./save_model/my_model"

CKPT_FILE = "../resnet_v1_50.ckpt"

LEARNING_RATE = 0.0001
STEPS = 500
N_CLASSES = 751

CHECKPOINT_EXCLUDE_SCOPES = 'Logits'

TRAINABLE_SCOPES = 'Logits'

def image_tranforms(img_data):
    img_data = tf.image.convert_image_dtype(img_data, dtype=tf.float32)
    img_data = tf.image.resize_images(img_data, [256, 128], method=1)
    img_data = tf.image.resize_image_with_crop_or_pad(img_data, 276, 148)
    img_data = tf.image.random_crop(img_data, [256, 128, 3])
    img_data = tf.image.random_flip_left_right(img_data)

    new_image_data1 = (img_data[:, :, 0] - 0.485) / 0.229
    new_image_data2 = (img_data[:, :, 1] - 0.456) / 0.224
    new_image_data3 = (img_data[:, :, 2] - 0.406) / 0.225
    img_data = tf.stack([new_image_data1, new_image_data2, new_image_data3], 2)

    return img_data
    # return tf.clip_by_value(img_data, 0.0, 1.0)

def parser(record):
    features = tf.parse_single_example(
        record,
        features={
            'image_raw' : tf.FixedLenFeature([], tf.string),
            'label' : tf.FixedLenFeature([], tf.int64)
        }
    )

    return tf.image.decode_jpeg(features['image_raw']), features['label']

def get_tuned_variables():
    exclusions = [scope.strip() for scope in CHECKPOINT_EXCLUDE_SCOPES.split(",")]
    variables_to_restore = []
    for var in slim.get_model_variables():
        excluded = False
        for exclusion in exclusions:
            if var.op.name.startswith(exclusion):
                excluded = True
                break
        if not excluded:
            variables_to_restore.append(var)
    return variables_to_restore

def get_trainable_variables():
    scopes = [scope.strip() for scope in TRAINABLE_SCOPES.split(",")]
    variables_to_train = []
    for scope in scopes:
        variables = tf.get_collection(tf.GraphKeys.TRAINABLE_VARIABLES, scope)
        variables_to_train.extend(variables)
    return variables_to_train

def main():
    train_dataset = tf.data.TFRecordDataset(train_tfrecords)
    val_dataset = tf.data.TFRecordDataset(val_tfrecords)
    train_dataset = train_dataset.map(parser)
    val_dataset = val_dataset.map(parser)

    train_dataset = train_dataset.map(
        lambda image, label: (image_tranforms(image), label)
    )
    val_dataset = val_dataset.map(
        lambda image, label: (image_tranforms(image), label)
    )

    train_dataset = train_dataset.shuffle(1000).batch(32)
    val_dataset = val_dataset.shuffle(1000).batch(300)

    train_iterator = train_dataset.make_initializable_iterator()
    val_iterator = val_dataset.make_initializable_iterator()
    img_data, label = train_iterator.get_next()
    val_img_data, val_label = val_iterator.get_next()

    images = tf.placeholder(tf.float32, [None, 256, 128, 3], name='input_images')
    labels = tf.placeholder(tf.int64, [None], name='labels')

    with slim.arg_scope(resnet_v1.resnet_arg_scope()):
        net, endpoints = resnet_v1.resnet_v1_50(images, num_classes=None)

    ##########################2019/4/29############################
    with tf.variable_scope("Logits"):
        net = tf.squeeze(net, axis=[1,2])
        net = slim.fully_connected(net, num_outputs=512, scope="fc1", normalizer_fn=slim.batch_norm,
                                   weights_initializer=tf.truncated_normal_initializer(mean=0, stddev=0.0625),
                                   activation_fn=None
                                  )
        net = slim.dropout(net, keep_prob=0.5, scope='dropout_scope')
        logits = slim.fully_connected(net, num_outputs=N_CLASSES, scope='fc2',normalizer_fn=slim.batch_norm,
                                      weights_initializer=tf.truncated_normal_initializer(mean=0, stddev=0.001),
                                      activation_fn=None)

    #     #######################################################

    trainable_variables = get_trainable_variables()

    cross_entropy = tf.nn.sparse_softmax_cross_entropy_with_logits(
        logits=logits, labels=labels
    )
    cross_entropy_mean = tf.reduce_mean(cross_entropy)
    tf.losses.add_loss(cross_entropy_mean)

    train_step = tf.train.RMSPropOptimizer(LEARNING_RATE).minimize(tf.losses.get_total_loss())

    with tf.name_scope('evaluation'):
        correct_prediction = tf.equal(tf.argmax(logits, 1), labels)
        evaluation_step = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))

    load_fn = slim.assign_from_checkpoint_fn(CKPT_FILE, get_tuned_variables(), ignore_missing_vars=True)

    saver = tf.train.Saver()
    with tf.Session() as sess:
        init = tf.global_variables_initializer()
        sess.run(init)
        sess.run([train_iterator.initializer, val_iterator.initializer])

        print("load the pretrained resnet model...")
        load_fn(sess)
        print("start training")
        for i in range(60):
            while True:
                try:
                    img_batch, label_batch = sess.run([img_data, label])              
                    sess.run(train_step, feed_dict={images: img_batch, labels: label_batch})
                except tf.errors.OutOfRangeError:
                    break
            sess.run(train_iterator.initializer)
            val_imgs, val_labels = sess.run([val_img_data, val_label])
            validation_accuracy = sess.run(evaluation_step, feed_dict={images: val_imgs,
                                                                       labels: val_labels})
            print("after %d train，the accuracy on val is %.3f" % (i+1, validation_accuracy))
            sess.run(val_iterator.initializer)

    print("finish")

if __name__ == '__main__':
    main()
`

**Comment by layumi** [comment]
@ln23415 , Sorry for the late response.
How about your learning rate? For the former layers in Resnet, I use 0.1*lrRate.
